pkg/portallocator/portallocator.go

```diff
@@ -236,20 +247,22 @@ func (pa *PortAllocator) DeAllocate(gs *agonesv1.GameServer) {
 
 // syncDeleteGameServer when a GameServer Pod is deleted
 // make the HostPort available
-func (pa *PortAllocator) syncDeleteGameServer(object interface{}) {
+func (pa *portAllocator) syncDeleteGameServer(object interface{}) {
 	if gs, ok := object.(*agonesv1.GameServer); ok {
 		pa.logger.WithField("gs", gs).Debug("Syncing deleted GameServer")
 		pa.DeAllocate(gs)
 	}
 }
 
-// syncAll syncs the pod, node and gameserver caches then
+// SyncAll syncs the pod, node and gameserver caches then
 // traverses all Nodes in the cluster and all looks at GameServers
 // and Terminating Pods values make sure those
 // portAllocations are marked as taken.
 // Locks the mutex while doing this.
 // This is basically a stop the world Garbage Collection on port allocations, but it only happens on startup.
-func (pa *PortAllocator) syncAll() error {
+//
+// SyncAll is exposed on the TestInterface only.
```

Comment:
@markmandel - I'm not an expert on how we use informers in our test code, but I'm wondering if there is a way to do it without needing to expose these functions outside of this package. Do you have any thoughts?

Reply:
FWIW, I did not spend long trying to figure that out, since a change in structure of allocating the informers earlier was considerably more invasive. I assume there may be a way to pull the informers out and pass them down, but maybe we can handle that as a follow-on? (Though it frankly seems somewhat low priority - I tried my best to ensure that it would take pretty explicit intent to call these outside the package.)

Reply:
In this particular instance, you could call `portallocator.Run(ctx)` in the test, since it spawns no worker threads, and isn't doing anything other than syncing the state. Even if it's just `context.Background()`.

That way you don't have to export the `syncAll()` function, `Run` is already public, and you don't need to export the `NodeSynced()` function either.

I think that should work!

Reply:
Ah, I missed that. Totally works, thanks!